src/freertos_drivers/esp_idf/EspIdfWiFi.cxx

```diff
@@ -0,0 +1,1355 @@
+        }
+        if (last)
+        {
+            last->ai_next = nullptr;
+        }
+        return *addr ? 0 : EAI_MEMORY;
+    }
+    if (!looking)
+    {
+        // Not currently looking for the service. Start looking.
+        mdns_scan(service);
+    }
+    return EAI_AGAIN;
+}
+
+//
+// EspIdfWiFi::mdns_scan()
+//
+void EspIdfWiFi::mdns_scan(const char *service)
+{
+    bool match = false;
+    OSMutexLock locker(this);
+    for (auto it = mdnsClientCache_.begin(); it != mdnsClientCache_.end(); ++it)
+    {
+        if ((*it).service_ == service)
+        {
+            match = true;
+        }
+    }
+    if (!match)
+    {
+        // Start looking for a new service.
+        mdnsClientCache_.emplace_back(std::move(std::string(service)));
+    }
+    if (!mdnsClientStarted_)
+    {
+        // Start the client if not already started.
+        mdnsClientStarted_ = true;
+        notify();
+        LOG(INFO, "wifi: mdns_scan() start.");
+    }
+    else if (!mdnsClientTrigRefresh_)
+    {
+        // Trigger the mDNS state machine to execute early. If a query is
+        // already taking place, a new one will start immediately following it.
+        // If a query it not taking place, one will be started immediately.
+        mdnsClientTrigRefresh_ = true;
+        CallbackExecutable *e =
+            new CallbackExecutable([this](){this->timer_.ensure_triggered();});
+        this->service()->executor()->add(e);
+        LOG(INFO, "wifi: mdns_scan() trigger.");
+    }
+}
+
+//
+// EspIdfWiFi::factory_default()
+//
+void EspIdfWiFi::factory_default()
+{
+    LOG(INFO, "wifi: factory_default()");
+    memset(&userCfg_, 0, sizeof(userCfg_));
+    userCfg_.magic_ = WIFI_CONFIG_INIT_MAGIC;
+    userCfg_.mode_ = WlanRole::AP_STA;
+    strncpy(userCfg_.ap_.ssid_, hostname_, MAX_SSID_SIZE);
+    userCfg_.ap_.ssid_[MAX_SSID_SIZE] = '\0';
+    strcpy(userCfg_.ap_.pass_, DEFAULT_PASSWORD);
+    userCfg_.ap_.sec_ = 2;
+    strcpy(userCfg_.sta_[0].ssid_, DEFAULT_STA_SSID);
+    strcpy(userCfg_.sta_[0].pass_, DEFAULT_PASSWORD);
+    userCfg_.sta_[0].sec_ = 2;
+}
+
+//
+// EspIdfWiFi::MDNSCacheItem::reset()
+void EspIdfWiFi::MDNSCacheItem::reset(void *handle)
+{
+    if (searchHandle_ && handle != searchHandle_)
+    {
+        mdns_query_async_delete((mdns_search_once_t*)searchHandle_);
+    }
+    searchHandle_ = handle;
+}
+
+//
+// EspIdfWiFi::mdns_start()
+//
+StateFlowBase::Action EspIdfWiFi::mdns_start()
+{
+    HASSERT(mdnsClientCache_.size() > 0);
+    if (mdnsAdvInhibitSta_)
+    {
+        mdns_adv_inhibit();
+        mdns_restore_sta();
+        mdnsAdvInhibitStaActive_ = true;
+    }
+    return call_immediately(STATE(mdns_query));
+}
+
+//
+// EspIdfWiFi::mdns_query()
+//
+StateFlowBase::Action EspIdfWiFi::mdns_query()
+{
+    OSMutexLock lock(this);
+    mdnsClientTrigRefresh_ = false;
+    for (auto it = mdnsClientCache_.begin(); it != mdnsClientCache_.end(); ++it)
+    {
+        std::string name;
```

Comment:
in cpp files we can likely remove the `std::` prefix as it should already be used by other libs, we can also consider adding a `using std;` at the top if needed

Reply:
I can't disagree, I similarly prefer explicit use statements where needed to restrict pollution of namespaces.